test/modules/generic-mqtt-tester/src/message_initiator.rs

```diff
@@ -91,10 +91,21 @@ impl MessageInitiator {
                 }
             };
 
-            self.report_message_sent(seq_num).await?;
+            let result = MessageTestResult::new(
+                self.settings.tracking_id(),
+                self.batch_id.to_string(),
+                seq_num,
+            );
+
+            let test_type = trc_client::TestType::Messages;
+            let created_at = chrono::Utc::now();
+            self.reporting_client
+                .report_result(SEND_SOURCE.to_string(), result, test_type, created_at)
+                .await
+                .map_err(MessageTesterError::ReportResult)?;
```

Comment:
Why did we take this out of the helper fn? Did the helper not seem worth it?

Reply:
borrow checker was complaining about the future still holds mutable reference to self. so I just inlined the whole thing

Reply:
hmm interesting, but okay works for me